### PR TITLE
Single in out

### PR DESCRIPTION
**Pull request by petermunnings**
There has been a lot of work done on this branch which needs to come back into master.  Once that has happened, I plan to work on developing a couple of circuits that can be called depending on your requirement.  All will be related to a balance update:  

* single payment (endBalance = startBalance + incomingPayment)
* single in and single out payment (startBal + incoming = endBal + outgoing)
* multiple in and multiple out payments

### CODE REVIEW

File: src/gadget.hpp [hunk 1]
```diff
@@ -46,16 +46,22 @@ class l_gadget : public gadget<FieldT> {
...
+    pb_variable_array<FieldT> intermediate_val4;
+    pb_variable_array<FieldT> intermediate_val5;
 
     std::shared_ptr<multipacking_gadget<FieldT> > unpack_inputs; /* multipacking gadget */
 
     std::shared_ptr<digest_variable<FieldT>> h1_var; /* H(R1) */
     std::shared_ptr<digest_variable<FieldT>> h2_var; /* H(R2) */
     std::shared_ptr<digest_variable<FieldT>> h3_var; /* H(R2) */
+    std::shared_ptr<digest_variable<FieldT>> h4_var; /* H(R2) */
+    std::shared_ptr<digest_variable<FieldT>> h5_var; /* H(R2) */
```
Comment (coeniebeyers) [inline src/gadget.hpp]: The comments might be confusing rather than helpfull, `h3_var` shouldn't be `H(R2)` etc

File: src/gadget.hpp [hunk 2]
```diff
@@ -46,16 +46,22 @@ class l_gadget : public gadget<FieldT> {
...
 
     std::shared_ptr<digest_variable<FieldT>> r1_var; /* R1 */
     std::shared_ptr<digest_variable<FieldT>> r2_var; /* R2 */
     std::shared_ptr<digest_variable<FieldT>> r3_var; /* R3 */
+    std::shared_ptr<digest_variable<FieldT>> r4_var; /* R3 */
+    std::shared_ptr<digest_variable<FieldT>> r5_var; /* R3 */
```
Comment (coeniebeyers) [inline src/gadget.hpp]: The comments might be confusing rather than helpfull, `r4_var` shouldn't be `R3` etc

Comment (petermunnings) [reply src/gadget.hpp]: fixed...

**Comment by coeniebeyers** [comment]
Excellent, merging

**Pull request merged**